src/utils/response.ts

```diff
@@ -41,3 +42,17 @@ export function appendHeader (res: ServerResponse, name: string, value: string):
 
   res.setHeader(name, current.concat(value))
 }
+
+export function isStream (data: any) {
+  return typeof data === 'object' && typeof data.pipe === 'function' && typeof data.on === 'function'
+}
+
+export function sendStream (res: ServerResponse, data: any) {
+  return new Promise((resolve, reject) => {
+    defer(() => {
```

Comment:
Using `defer` was mostly a performance hack for Node.js writable. I guess with pipes, we can start without it and see if it can really improve perf. (You can try with `profile` script if interested to investigate)

Reply:
> Using `defer` was mostly a performance hack for Node.js writable. I guess with pipes, we can start without it and see if it can really improve perf. (You can try with `profile` script if interested to investigate)

It does not change anything, but I will remove it as it an unnecessary operation. Would you like I add a profile:stream script ?